### PR TITLE
ci: Replace python go proto script with ci target

### PR DESCRIPTION
**Pull request by phlax**
This replaces the current python script to build the go protos with a bash ci target, and removes the git logic which has been shifted downstream

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by alecholmez — COMMENTED** [review]
This was migrated to go-control-plane 👍

**Comment by phlax** [comment]
not sure why ci is failing. I tested it locally and got it working, then pushed and it failed due to RBE, so reproed locally with RBE and fixed, but its still failing 8/

**Comment by phlax** [comment]
debugging CI fail further locally - afaict the issue is because the api is being built previously, and as a result its using the cache and not downloading the go files - it may be related to https://github.com/bazelbuild/bazel/issues/13882 - i think it probably passed previously due to building the go files first

adding `--remote_download_outputs=all` seems to resolve and didnt seem to have a huge perf penaltly when testing locally

**Comment by phlax** [comment]
this needs https://github.com/envoyproxy/go-control-plane/pull/701 in place before it can land

im also wondering whether we should add a stub in the python script to raise an error with a "Please use ci/do_ci.sh api.go instead" message

cc @jamesmulcahy

**Comment by phlax** [comment]
updated the script to message and `exit(1)`

also added an `ENVOY_API_ONLY` env var to optionally skip building the go protobufs

**Comment by phlax** [comment]
@alecholmez there is a bit of a chicken/egg situation here - we kinda need to simultaneously update downstream

i guess its possible we could do it in steps, ill raise a PR to update and use the new target now

**Comment by phlax** [comment]
> ~there is a bit of a chicken/egg situation here~

apologies for confusion - thinking further - downstream has its own implementation so this just needs to land first

File: tools/api/generate_go_protobuf.py [hunk 1]
```diff
@@ -1,164 +1,13 @@
...
-    git(repo, 'push', 'origin', BRANCH)
-
-
-def updated(repo):
-    return len([
-        f for f in git(repo, 'diff', 'HEAD', '--name-only').splitlines() if f != 'envoy/COMMIT'
-    ]) > 0
+def main():
+    # TODO(phlax): remove this 12/23
+    sys.stderr.write("This script has been removed, please use `ci/do_ci.sh api.go` instead\n")
```
Comment (alecholmez) [inline tools/api/generate_go_protobuf.py:8]: Should we just delete this script? Less cruft n the tools folder

**Review by alecholmez — APPROVED** [review]
Looks good to me!

File: tools/api/generate_go_protobuf.py (hunk 1 above)
Comment (phlax) [reply tools/api/generate_go_protobuf.py:8]: im a little concerned that others may be using it so would rather add this ~deprecation/removal notice for now

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**